autogluon/utils/tabular/ml/trainer/abstract_trainer.py

```diff
@@ -1242,6 +1246,93 @@ def get_models_info(self, models=None):
+        if models_to_keep is not None:
+            if not isinstance(models_to_keep, list):
+                models_to_keep = [models_to_keep]
+            minimum_model_set = set()
+            for model in models_to_keep:
+                minimum_model_set.update(self.get_minimum_model_set(model))
+            minimum_model_set = list(minimum_model_set)
+            models_to_remove = [model for model in self.get_model_names_all() if model not in minimum_model_set]
+        elif models_to_delete is not None:
+            if not isinstance(models_to_delete, list):
+                models_to_delete = [models_to_delete]
+            minimum_model_set = set(models_to_delete)
+            minimum_model_set_orig = copy.deepcopy(minimum_model_set)
+            for model in models_to_delete:
+                minimum_model_set.update(nx.algorithms.dag.descendants(self.model_graph, model))
+            if not allow_delete_cascade:
+                if minimum_model_set != minimum_model_set_orig:
+                    raise AssertionError('models_to_delete contains models which cause a delete cascade due to other models being dependent on them. Set allow_delete_cascade=True to enable the deletion.')
+            minimum_model_set = list(minimum_model_set)
+            models_to_remove = [model for model in self.get_model_names_all() if model in minimum_model_set]
+        else:
+            raise ValueError('Exactly one of [models_to_keep, models_to_delete] must be set.')
+
+        if dry_run:
+            logger.log(30, f'Dry run enabled, AutoGluon would have deleted the following models: {models_to_remove}')
+            if delete_from_disk:
+                for model in models_to_remove:
+                    model = self.load_model(model)
+                    logger.log(30, f'\tDirectory {model.path} would have been deleted.')
+            logger.log(30, f'To perform the deletion, set dry_run=False')
+            return
+
+        self.model_graph.remove_nodes_from(models_to_remove)
+        for model in models_to_remove:
+            if model in self.models:
+                self.models.pop(model)
+
+        models_kept = self.get_model_names_all()
+        # TODO: Refactor this part, link models_level to model_graph
+        for key in self.models_level:
+            for level in self.models_level[key]:
+                self.models_level[key][level] = [model for model in self.models_level[key][level] if model in models_kept]
+
+        if self.model_best is not None and self.model_best not in models_kept:
+            try:
+                self.model_best = self.get_model_best()
+            except AssertionError:
+                self.model_best = None
+
+        # TODO: Delete from all the other model dicts
+        self.save()
+        if delete_from_disk:
+            for model in models_to_remove:
+                model = self.load_model(model)
+                model.delete_from_disk()
```

Comment:
add logging statements here to clarify deletion has happened

Reply:
Logging statement is present in model.delete_from_disk()